lib/omniauth/strategies/openid_connect.rb

```diff
@@ -185,6 +182,14 @@ def public_key
 
       private
 
+      def only_id_token?
+        if options.response_type.is_a?(Array)
```

Comment:
please consider following:

```ruby
types = Array(options.response_type)
types.first.to_s == 'id_token' && types.size == 1
```

I'd prefer to check length of array instead of using `#one?` which is intended to check how much elements in array are truthy, not for measuring the length of enumerable.

Reply:
I changed this method.

Reply:
Why do you insist on this complex logic instead of memoizing of normalized `response_types` array?

Reply:
In the tests we assign response types like that: `strategy.options.response_type = :id_token` and I don't know if there are users who dynamically change the `response_type`. If so, I'm not sure if memoization could work.

Reply:
> dynamically change the response_type

what? why?

Reply:
I don't if there is such a case. I can memoize it.

Reply:
Do you have experience of changing settings in runtime? Can you describe why anybody would do that?

Sorry but this doesn't make sense to me.

Reply:
No, I don't have.